File: iris-ld/iris-ld-jsonld/src/main/java/city/smartb/iris/jsonld/serializer/JsonLdObjectSerializer.java
package city.smartb.iris.jsonld.serializer;

import city.smartb.iris.jsonld.JsonLdObject;
import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
import com.fasterxml.jackson.databind.ser.std.StdSerializer;

import java.io.IOException;
import java.util.Map;

public class JsonLdObjectSerializer extends StdSerializer<JsonLdObject> {

    public JsonLdObjectSerializer() {
        this(null);
    }

    public JsonLdObjectSerializer(Class<JsonLdObject> t) {
        super(t);
    }

    @Override
    public void serialize(JsonLdObject value, JsonGenerator jgen, SerializerProvider provider) throws IOException {
        jgen.writeObject(value.asJson());
    }
}

File: iris-ld/iris-ld-jsonld/src/main/kotlin/city/smartb/iris/ld/jsonld/serializer/JsonLdObjectSerializer.kt
package city.smartb.iris.ld.jsonld.serializer

import city.smartb.iris.ld.jsonld.JsonLdObject
import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.ser.std.StdSerializer
import java.io.IOException

class JsonLdObjectSerializer @JvmOverloads constructor(t: Class<JsonLdObject?>? = null) :
    StdSerializer<JsonLdObject>(t) {
    @Throws(IOException::class)
    override fun serialize(value: JsonLdObject, jgen: JsonGenerator, provider: SerializerProvider) {
        jgen.writeObject(value.asJson())
    }
}
